### PR TITLE
Vhost rxtx counter (+ small fixes)

### PR DESCRIPTION
**Pull request by outscale-jju**

### CODE REVIEW

File: include/packetgraph/vhost.h [hunk 1]
```diff
@@ -66,4 +66,20 @@ const char *pg_vhost_socket_path(struct pg_brick *brick,
  */
 void pg_vhost_stop(void);
 
+/**
+ * Get RX count.
+ * This function is thread-safe.
+ * @param: brick pointer to the vhost brick
+ * @return: number of bytes received by the VM (PG -> VM)
+ */
+uint64_t pg_vhost_rx(struct pg_brick *brick);
```
Comment (outscale-mgo) [inline include/packetgraph/vhost.h]: And what about the posibility to do the same with brick nix ?
I would add a brick_rx/tx in pg_brick_ops instead

File: src/vhost.c [hunk 2]
```diff
@@ -162,9 +165,15 @@ static int vhost_poll(struct pg_brick *brick, uint16_t *pkts_cnt,
 
 	pkts_mask = pg_mask_firsts(count);
 	ret = pg_brick_side_forward(s, state->output,
-				 state->in, pkts_mask, errp);
+				    state->in, pkts_mask, errp);
 	pg_packets_free(state->in, pkts_mask);
```
Comment (outscale-mgo) [inline src/vhost.c]: here you free the packets

File: src/vhost.c [hunk 3]
```diff
@@ -162,9 +165,15 @@ static int vhost_poll(struct pg_brick *brick, uint16_t *pkts_cnt,
...
 
+	/* count rx bytes: burst is packed so we can directly iterate */
+	in = state->in;
```
Comment (outscale-mgo) [inline src/vhost.c]: here you set in as state->in

File: src/vhost.c [hunk 4]
```diff
@@ -162,9 +165,15 @@ static int vhost_poll(struct pg_brick *brick, uint16_t *pkts_cnt,
...
+	for (int i = 0; i < count; i++)
+		rx_bytes += rte_pktmbuf_pkt_len(in[i]);
```
Comment (outscale-mgo) [inline src/vhost.c]: and here you count in aka state->in that you have just free...

Comment (outscale-mgo) [inline src/vhost.c]: I don't understand why you are counting len, after the bust has been make ?

File: src/vhost.c [hunk 5]
```diff
@@ -275,6 +286,22 @@ const char *pg_vhost_socket_path(struct pg_brick *brick,
...
+{
+	struct pg_vhost_state *state;
+
+	state = pg_brick_get_state(brick, struct pg_vhost_state);
+	return rte_atomic64_read(&state->rx_bytes);
+}
+
+static uint64_t tx_bytes(struct pg_brick *brick)
+{
+	struct pg_vhost_state *state;
```
Comment (outscale-mgo) [inline src/vhost.c]: I think you could just have write:
``` return rte_atomic64_read(&pg_brick_get_state(brick, struct pg_vhost_state)->tx_bytes); ```

**Comment by outscale-mgo** [comment]
"vhost: support rx_bytes and tx_bytes operations" should be after "core: new operation for counting transfered bytes"

File: src/vhost.c (hunk 2 above)
Comment (outscale-jju) [inline src/vhost.c]: ok that's a mistake

File: src/vhost.c (hunk 4 above)
Comment (outscale-jju) [inline src/vhost.c]: before or after does not have any matter

File: src/vhost.c (hunk 5 above)
Comment (outscale-jju) [inline src/vhost.c]: ok but more obscure

File: src/vhost.c (hunk 4 above)
Comment (outscale-mgo) [inline src/vhost.c]: if you have a vtep on the way, packets size will increase and the result wont be the same

Comment (outscale-jju) [inline src/vhost.c]: true, fixed anyway

**Pull request closed without merging**